### ISSUE TITLE
[ARC E2E] Huggingface accuracy failed

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#911]
### 🐛 Describe the bug

Accuracy failed for key name albert.embeddings.token_type_embeddings.weight.grad
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/mengfeil/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/mengfeil/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Category | Model | Accuracy
-- | -- | --
huggingface_amp_bf16_training | AlbertForMaskedLM | fail_accuracy
huggingface_amp_fp16_training | AlbertForMaskedLM | fail_accuracy
huggingface_bfloat16_training | AlbertForMaskedLM | fail_accuracy
huggingface_float16_training | AlbertForMaskedLM | fail_accuracy
huggingface_float32_training | AlbertForMaskedLM | fail_accuracy
huggingface_amp_bf16_training | AlbertForQuestionAnswering | fail_accuracy
huggingface_amp_fp16_training | AlbertForQuestionAnswering | fail_accuracy
huggingface_bfloat16_training | AlbertForQuestionAnswering | fail_accuracy
huggingface_float16_training | AlbertForQuestionAnswering | fail_accuracy
huggingface_float32_training | AlbertForQuestionAnswering | fail_accuracy

</body>

</html>

### Versions

torch-xpu-ops: 7e3d00acea9f0d3728048a5b2743de20d55c64ba
pytorch: 0d1d69fd25fdc096763bfe85f4d379e27ea1c9f8
device: ARC 24.04
driver: 24.31.30508.7

**Comment by mengfei25** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/mengfeil/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/mengfeil/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">


Category | Model | Ubuntu 24.04 | Ubuntu 22.04
-- | -- | -- | --
huggingface_amp_bf16_training | AlbertForMaskedLM | fail_accuracy | pass
huggingface_bfloat16_training | AlbertForMaskedLM | fail_accuracy | pass
huggingface_float32_training | AlbertForMaskedLM | fail_accuracy | pass
huggingface_float32_training | AlbertForQuestionAnswering | fail_accuracy | pass



</body>

</html>

**Comment by jianyizh** [comment]
@mengfei25 
I can reproduce on ubuntu 22, for example, AlbertForMaskedLM fp16 training, I get 
W0925 06:28:19.597000 315406 site-packages/torch/_dynamo/utils.py:1723] Similarity score=0.9181798100471497
E0925 06:28:19.598000 315406 site-packages/torch/_dynamo/utils.py:1674] Accuracy failed for key name albert.embeddings.token_type_embeddings.weight.grad. The similarity score is exactly the same as your test log.
However, If I modify the patch and compare the result directly, it can pass. You do not need put fp64_outputs on xpu, and you can put new_result and correct_result on cpu. https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/common.py#L2898

**Comment by jianyizh** [comment]
cuda also fails on cosine similarity
cuda train AlbertForMaskedLM
W0927 14:50:50.191000 3628133 /mnt/tmp/home/sparse/miniforge3/envs/jianyi/lib/python3.10/site-packages/torch/_inductor/debug.py:434] [3/0_1] AlbertForMaskedLM__0_forward_1 debug trace: /home/sparse/jianyi/pytorch/inductor_log/huggingface/AlbertForMaskedLM/training/amp_fp16/22/torch_compile_debug/torch_compile_debug/run_2024_09_27_14_50_33_656959-pid_3628133/torchinductor/AlbertForMaskedLM__0_forward_1.0
W0927 14:51:04.399000 3628133 /mnt/tmp/home/sparse/miniforge3/envs/jianyi/lib/python3.10/site-packages/torch/_inductor/debug.py:434] [3/0_1] AlbertForMaskedLM__0_backward_3 debug trace: /home/sparse/jianyi/pytorch/inductor_log/huggingface/AlbertForMaskedLM/training/amp_fp16/22/torch_compile_debug/torch_compile_debug/run_2024_09_27_14_50_33_656959-pid_3628133/torchinductor/AlbertForMaskedLM__0_backward_3.1
W0927 14:51:05.220000 3628133 /mnt/tmp/home/sparse/miniforge3/envs/jianyi/lib/python3.10/site-packages/torch/_dynamo/utils.py:1723] Similarity score=0.6160069704055786
E0927 14:51:05.221000 3628133 /mnt/tmp/home/sparse/miniforge3/envs/jianyi/lib/python3.10/site-packages/torch/_dynamo/utils.py:1674] Accuracy failed for key name albert.embeddings.token_type_embeddings.weight.grad
fail_accuracy

**Comment by chuanqi129** [comment]
@jianyizh does it mean both xpu and cuda will fail on cosine similarity for those models?

**Comment by jianyizh** [comment]
> @jianyizh does it mean both xpu and cuda will fail on cosine similarity for those models?

yes, and they can pass on the normal accuracy test with fp64 patch corrected

**Comment by riverliuintel** [comment]
Clean issues. Use PT2.6 based on Ubuntu 24.10 fully testing and submit new issues.
